fedimint-server/src/consensus/server.rs

```diff
@@ -822,13 +824,25 @@ async fn submit_module_consensus_items(
             while !task_handle.is_shutting_down() {
                 let mut dbtx = db.begin_transaction_nc().await;
 
-                for (instance_id, _, module) in modules.iter_modules() {
-                    let module_consensus_items = module
-                        .consensus_proposal(
-                            &mut dbtx.to_ref_with_prefix_module_id(instance_id).into_nc(),
-                            instance_id,
-                        )
-                        .await;
+                for (module_id, kind, module) in modules.iter_modules() {
+                    let module_consensus_items = tokio::time::timeout(
+                        CONSENSUS_PROPOSAL_TIMEOUT,
+                        module.consensus_proposal(
+                            &mut dbtx.to_ref_with_prefix_module_id(module_id).into_nc(),
+                            module_id,
+                        ),
+                    )
+                    .await;
```

Comment:
There are some theoretical concerns around being able to make guardians not propose by third parties somehow making `consensus_proposal` take longer than the timeout. But the practical concerns far outweigh this.

Reply:
This can always be solved by doing slow operations on background thread or something like that, exactly like in #4916 .

And since modules are polled for consensus proposal serially RN, a slow modules is making consensus proposal slower for all other modules. @elsirion 

We should probably spawn a separate loop for every module instance. @joschisan what do you think?